.storybook/webpack.config.js

```diff
@@ -6,31 +6,58 @@
 const path = require('path')
 const webpack = require('webpack')
 const fs = require('fs')
+const config = require('../build/commands/lib/config')
+const genTsConfig = require('../build/commands/lib/genTsConfig')
 const { fallback, provideNodeGlobals } = require('../components/webpack/polyfill')
+const generatePathMap = require('../components/webpack/path-map')
 
-const buildConfigs = ['Component', 'Static', 'Debug', 'Release']
-const extraArchitectures = ['arm64', 'x86']
+// Choose which brave-core build directory to look for pre-compiled
+// resource dependencies:
+// 1. Default for local builds for the actual platform / architecture
+// 2. platform / architecture overriden by environment variables
+// 3. most recently built - this caters to the common scenario when a
+// non-standard target has been built but no arguments are provided to storybook
+config.update({
+  target_arch: process.env.TARGET_ARCH,
+  target_os: process.env.TARGET_OS,
+  target_environment: process.env.TARGET_ENVIRONMENT,
+  target: process.env.TARGET
+})
 
-function getBuildOuptutPathList(buildOutputRelativePath) {
+let outputPath = config.outputDir
+
+function getBuildOuptutPathList() {
   return buildConfigs.flatMap((config) => [
-    path.resolve(__dirname, `../../out/${config}/${buildOutputRelativePath}`),
+    path.resolve(__dirname, `../../out/${config}`),
     ...extraArchitectures.map((arch) =>
       path.resolve(
         __dirname,
-        `../../out/${config}_${arch}/${buildOutputRelativePath}`
+        `../../out/${config}_${arch}`
       )
     )
   ])
 }
 
-const genFolder = getBuildOuptutPathList('gen')
-  .filter(a => fs.existsSync(a))
-  .sort((a, b) => fs.statSync(b).mtime - fs.statSync(a).mtime)[0]
-if (!genFolder) {
-  throw new Error("Failed to find build output folder!")
+if (fs.existsSync(config.outputDir)) {
```

Comment:
I would use `outputPath` instead of `config.outputDir` here. We already have it in the variable, and it's also what's used in the `if` body.